template/src/navigation/NavigationService.js

```diff
@@ -1,18 +1,24 @@
-import * as React from 'react'
 import {StackActions} from '@react-navigation/native'
+import {createNavigationContainerRef} from '@react-navigation/native';
 
-export const navigationRef = React.createRef()
+export const navigationRef = createNavigationContainerRef()
 
 export function navigate(name, params) {
-  navigationRef.current?.navigate(name, params)
+  if (navigationRef.isReady()) {
+    navigationRef.navigate(name, params)
+  }
 }
 
-export const checkRouteOrigin = () => navigationRef.current.getRootState().routeNames[0]
+export const checkRouteOrigin = () => navigationRef.getRootState().routeNames[0]
 
 export function navigationPop() {
-  navigationRef.current?.dispatch(StackActions.pop(1))
+  if (navigationRef.isReady()) {
+    navigationRef.dispatch(StackActions.pop(1))
```

Comment:
Because you are updating this function, can we make it able to pass number of times we want to pop, with default value `1`, like 
```
export function navigationPop(time = 1) {
...
navigationRef.dispatch(StackActions.pop(time))
...
```

Reply:
updated